### PR TITLE
Functional tests - Add new test 'BO > Product V2 > Details tab'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new test 'BO > Product V2 > Details tab'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | no need
| UI Tests          | Nightly => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/6261807640
| Fixed issue or discussion?     | no
| Related PRs       | no
| Sponsor company   |

### CODE REVIEW

File: tests/UI/data/faker/product.ts [hunk 1]
```diff
@@ -135,6 +156,18 @@ export default class ProductData {
...
+
+    /** @type {string|null} UPC of the product */
+    this.UPC = productToCreate.UPC || null;
+
+    /** @type {string|null} EAN13 of the product */
+    this.EAN13 = productToCreate.EAN13 || null;
+
+    /** @type {string|null} ISBN of the product */
+    this.ISBN = productToCreate.ISBN || null;
+
```
Comment (boubkerbribri) [inline tests/UI/data/faker/product.ts]: Even if there are abbreviation, it should not be uppercase as other vars are on camelCase

File: tests/UI/pages/BO/catalog/productsV2/add/detailsTab.ts [hunk 2]
```diff
@@ -100,14 +205,169 @@ class DetailsTab extends BOBasePage {
...
+
   /**
    * @param page {Page}
    * @param inputName {string}
    */
   async getValue(page: Page, inputName: string): Promise<string> {
     switch (inputName) {
       case 'condition':
-        return page.$eval(this.productConditionInput, (node: HTMLSelectElement) => node.value);
+        return page.$eval(this.productConditionSelect, (node: HTMLSelectElement) => node.value);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/detailsTab.ts]:
```suggestion
        return page
          .locator(this.productConditionSelect)
          .evaluate((el) => el.value);
```

File: tests/UI/pages/BO/catalog/productsV2/add/detailsTab.ts [hunk 3]
```diff
@@ -100,14 +205,169 @@ class DetailsTab extends BOBasePage {
...
+   * @returns {Promise<void>}
+   */
+  async addNewFile(page: Page, productData: ProductData): Promise<void> {
+    for (let i: number = 0; i < productData.files.length; i++) {
+      await this.waitForSelectorAndClick(page, this.addNewFileButton);
+
+      await this.waitForVisibleSelector(page, this.createFileFrame);
+
+      const newFileFrame: Frame | null = await page.frame({name: 'modal-create-product-attachment-iframe'});
+      await expect(newFileFrame).to.be.not.null;
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/productsV2/add/detailsTab.ts]:
```suggestion
      const newFileFrame: Frame | null = page.frame({name: 'modal-create-product-attachment-iframe'});
      expect(newFileFrame).to.be.not.eq(null);
```

File: tests/UI/data/faker/product.ts (hunk 1 above)
Comment (Progi1984) [reply tests/UI/data/faker/product.ts]: :+1: for this.

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: tests/UI/data/faker/product.ts (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/data/faker/product.ts]: Done

File: tests/UI/pages/BO/catalog/productsV2/add/detailsTab.ts (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/catalog/productsV2/add/detailsTab.ts]: done

File: tests/UI/pages/BO/catalog/productsV2/add/detailsTab.ts (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/catalog/productsV2/add/detailsTab.ts]: Done

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**